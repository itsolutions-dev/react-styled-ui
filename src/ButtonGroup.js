import styledComponent, { css } from './styledComponent';

const ButtonGroup = styledComponent('div', css`
  display: flex;
  flex-direction: ${props => props.vertical === true ? 'column' : 'row'};

  > button {
    margin: 0;
  }
`);

export default ButtonGroup;
