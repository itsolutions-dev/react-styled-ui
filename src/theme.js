export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
