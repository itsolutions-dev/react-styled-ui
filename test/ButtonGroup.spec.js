import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import ButtonGroup from '../src/ButtonGroup';

describe('ButtonGroup', () => {
  test('should render a ButtonGroup', () => {
    const component = ReactTestRenderer.create(<ButtonGroup theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag', () => {
    const component = ReactTestRenderer.create(<ButtonGroup theme={theme} />).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should horizontal by default', () => {
    const component = ReactTestRenderer.create(
      <ButtonGroup
        theme={theme}
      />,
    );
    expect(component).toHaveStyleRule('flex-direction', 'row');
  });

  test('should be vertical', () => {
    const component = ReactTestRenderer.create(
      <ButtonGroup
        theme={theme}
        vertical
      />,
    );
    expect(component).toHaveStyleRule('flex-direction', 'column');
  });

  test('should have child button with margin 0', () => {
    const component = ReactTestRenderer.create(
      <ButtonGroup
        theme={theme}
        vertical
      />,
    );
    expect({
      component,
      modifier: '> button',
    }).toHaveStyleRule('margin', '0');
  });
});
