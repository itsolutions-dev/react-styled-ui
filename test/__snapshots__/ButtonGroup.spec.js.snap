// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render a ButtonGroup 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c0 > button {
  margin: 0;
}

<div
  className="c0"
/>
`;
